internal/gengapic/client_init.go

```diff
@@ -139,193 +139,405 @@ func (g *generator) clientOptions(serv *descriptor.ServiceDescriptorProto, servN
-		p("connPool gtransport.ConnPool")
-		p("")
+	// Fields
+	p("// The internal transport-dependent client.")
+	p("internalClient internal%sClient", servName)
+	p("")
+	p("// The call options for this service.")
+	p("CallOptions *%sCallOptions", servName)
+	p("")
 
-		p("// flag to opt out of default deadlines via %s", disableDeadlinesVar)
-		p("disableDeadlines bool")
+	// Need to keep for back compat
+	if hasRPCForLRO {
+		p("// LROClient is used internally to handle long-running operations.")
+		p("// It is exposed so that its CallOptions can be modified if required.")
+		p("// Users should not Close this client.")
+		p("LROClient *lroauto.OperationsClient")
 		p("")
+		g.imports[pbinfo.ImportSpec{Name: "lroauto", Path: "cloud.google.com/go/longrunning/autogen"}] = true
+	}
 
-		p("// The gRPC API client.")
-		p("%s %s.%sClient", grpcClientField(servName), imp.Name, serv.GetName())
-		p("")
+	p("}")
+	p("")
 
-		if hasRPCForLRO {
-			p("// LROClient is used internally to handle longrunning operations.")
-			p("// It is exposed so that its CallOptions can be modified if required.")
-			p("// Users should not Close this client.")
-			p("LROClient *lroauto.OperationsClient")
-			p("")
+	g.clientUtilities(serv, servName)
+}
 
-			g.imports[pbinfo.ImportSpec{Name: "lroauto", Path: "cloud.google.com/go/longrunning/autogen"}] = true
-		}
+func (g *generator) clientUtilities(serv *descriptor.ServiceDescriptorProto, servName string) error {
 
-		if g.hasLROMixin() {
-			p("operationsClient longrunningpb.OperationsClient")
-			p("")
+	p := g.printf
 
-			g.imports[pbinfo.ImportSpec{Name: "longrunningpb", Path: "google.golang.org/genproto/googleapis/longrunning"}] = true
+	// Close
+	p("func (c *%sClient) Close () error {", servName)
+	p("    return c.internalClient.Close()")
+	p("}")
+	p("")
+
+	// Need a thin wrapper around
+	for _, m := range serv.Method {
+		inType := g.descInfo.Type[m.GetInputType()]
+		inSpec, err := g.descInfo.ImportSpec(inType)
+		if err != nil {
+			return err
+		}
+		outType := g.descInfo.Type[m.GetOutputType()]
+		outSpec, err := g.descInfo.ImportSpec(outType)
+		if err != nil {
+			return err
 		}
 
-		if g.hasIAMPolicyMixin() {
+		if m.GetOutputType() == emptyType {
+			p("func (c *%sClient) %s(ctx context.Context, req *%s.%s, opts ...gax.CallOption) error {",
+				servName,
+				m.GetName(),
+				inSpec.Name,
+				inType.GetName())
+			p("    var opts gax.CallOption")
+			g.appendCallOpts(m)
+			p("    c.internalClient.%s(ctx, req, opts)", m.GetName())
+			p("}")
+			p("")
+			continue
+		}
 
-			p("iamPolicyClient iampb.IAMPolicyClient")
+		if pf, err := g.pagingField(m); err != nil {
+			return err
+		} else if pf != nil {
+			iter, err := g.iterTypeOf(pf)
+			if err != nil {
+				return err
+			}
+			p("func (c *%sClient) %s(ctx context.Context, req *%s.%s, opts ...gax.CallOption) *%s {",
+				servName,
+				m.GetName(),
+				inSpec.Name,
+				inType.GetName(),
+				iter.iterTypeName)
+			p("    var opts gax.CallOption")
```

Comment:
This is an unnecessary variable declaration, there is already a `opts gax.CallOption` defined as a parameter.